### PR TITLE
Adapt webservice for extract

### LINKED ISSUE #1181: New with images parameter
__GetExtractById Service:__
---

New withimages parameter

**OLD:**

`https://example.com/oereb/extract/xml/WITHIMAGES/CH887722167773`

**NEW:**

`https://example.com/oereb/extract/CH887722167773?WITHIMAGES=true`
`https://example.com/oereb/extract/CH887722167773?WITHIMAGES=false`
`https://example.com/oereb/extract/CH887722167773?` => like WITHIMAGES=false

### LINKED ISSUE #1182: New signed parameter parameter
__GetExtractById Service:__
---

New signed parameter parameter

**OLD:**

`https://example.com/oereb/extract/xml/SIGNED/CH887722167773`

**NEW:**

`https://example.com/oereb/extract/CH887722167773?SIGNED=true`
`https://example.com/oereb/extract/CH887722167773?SIGNED=false`
`https://example.com/oereb/extract/CH887722167773?` => like SIGNED=false

### LINKED ISSUE #1183: New geometry parameter
__GetExtractById Service:__
---

New geometry parameter

**OLD:**

`https://example.com/oereb/extract/xml/GEOMETRY/CH887722167773`

**NEW:**

`https://example.com/oereb/extract/CH887722167773?GEOMETRY=true`
`https://example.com/oereb/extract/CH887722167773?GEOMETRY=false`
`https://example.com/oereb/extract/CH887722167773?` => like GEOMETRY=false

### LINKED ISSUE #1185: By IDENTDN and NUMBER (Type B)
__GetExtractById Service:__
---

By IDENTDN and NUMBER (Type B)

**OLD:**

`https://example.com/oereb/extract/xml/{STRING}/{STRING}`

`https://example.com/oereb/extract/xml/BE0200000332/100`

**NEW:**

`https://example.com/oereb/extract/xml?{IDENTDN=STRING}&{NUMBER=STRING}`

`https://example.com/oereb/extract/xml?IDENTDN=BE0200000332&NUMBER=100`

Mandatory: IDENTDN, NUMBER

### LINKED ISSUE #1186: By EGRID (Type A)
__GetExtractById Service:__
---

By EGRID (Type A)

**OLD:**

`https://example.com/oereb/extract/xml/{STRING}`

`https://example.com/oereb/extract/xml/CH887722167773`

**NEW:**

`https://example.com/oereb/extract/xml?{EGRID=STRING}`

`https://example.com/oereb/extract/xml?EGRID=CH887722167773`

Mandatory: EGRID

### LINKED ISSUE #1227: Reduced becomes the new standard DATA-extract
---
the attribute 'isReduced' is removed as is the FLAVOUR parameter

Should we keep the parameter in pyramid_oereb?

##### Consequences:
---
* Code can be cleaned

### PR DESCRIPTION
**Pull request by kdeininger**
Closes #1181
Closes #1182
Closes #1183
Closes #1185
Closes #1186
Closes #1227

### CODE REVIEW

**Review by vvmruder — COMMENTED** [review]
Only one suggestion. Otherwise this is good and reviewed.

File: pyramid_oereb/views/webservice.py [hunk 1]
```diff
@@ -334,26 +334,17 @@ def __validate_extract_params__(self):
...
-            raise HTTPBadRequest('Invalid flavour: {0}'.format(extract_flavour))
-
         # Get and check format
         extract_format = self.__validate_format_param__(self._EXTRACT_FORMATS)
 
         # With geometry?
         with_geometry = False
         user_requested_geometry = False
-        if self._request.matchdict.get('param1').lower() == 'geometry':
+        if self._request.params.get('GEOMETRY', 'false').lower() == 'true':
```
[the remark this answers is not in the document]
Comment (kdeininger) [reply pyramid_oereb/views/webservice.py:343]: This is already done here:
https://github.com/openoereb/pyramid_oereb/blob/cf298874cd9a897f99a127579b322a02727f772e/pyramid_oereb/views/webservice.py#L41

There are also tests for case insensitive parameters:
https://github.com/openoereb/pyramid_oereb/blob/cf298874cd9a897f99a127579b322a02727f772e/tests/webservice/test_case_insensitivity.py

[the remark this answers is not in the document]
Comment (vvmruder) [reply pyramid_oereb/views/webservice.py:343]: But I was not able to get extract with geometries when I use `geometries=true` as parameter.

I use this:
http://localhost:6543/oereb/extract/xml?EGRID=CH113928077734&GEOMETRY=true

and it contains Geometries.

I use:
http://localhost:6543/oereb/extract/xml?EGRID=CH113928077734&geometry=true

And it does not. What am I missing?

[the remark this answers is not in the document]
Comment (kdeininger) [reply pyramid_oereb/views/webservice.py]: My fault... I have used the wrong dict :see_no_evil: 

Should be fixed now.

**Review by vvmruder — APPROVED** [review]
Now it looks good. Thank you for fixing.

**Review by voisardf — APPROVED** [review]
Sorry, I forgot to send the approval! 🙈 thanks and go for it!

**Pull request merged**